optas/solver.py

```diff
@@ -89,6 +91,7 @@ def reset_parameters(self, p):
 
         """
         self.p = self.opt.parameters.dict2vec(p)
+        self._p_dict = self.opt.parameters.vec2dict(self.p)
```

Comment:
Actually, i need to.
Because otherwise if the user does not initialize the parameter to zero, they won't be part of the _p_dict and then it will fail once we try to merge.
This way, even if the user does not initialize them the first line will do it and the second will just store it as a dictionary.
In summary, p is different from self._p_dict

Reply:
Yes, but `vec2dict` assumes the input is a vector. The input here is a dictionary - this will fail. Handling the case when the user decides not to provide the parameters can be handled in `_solve`.

Reply:
First, it won't fail, because I tested it. And it doesn't fail.

The p is a dictionary. The first dict2vec turns saves it as a self.p which is a vector. (but with all the parameters initialised.
The self.p is again a vector, so I just turn it again to a dictionary self._p_dict.

Maybe I can change the names to make it more clear.

I don't agree that the default parameters should be handled by the _solve(). This is a already quite cluttered. And as it is know the reset_parameters is handling both cases of of setting the default parameters.

Reply:
Aaahh, sorry. I was reading this as if it were
```python
self._p_dict = self.opt.parameters.vec2dict(p)
```
which would fail. Yes, you're correct - this would pass fine as you have it.

Reply:
Ok, all good!